cmd/minikube/cmd/config/profile.go

```diff
@@ -71,3 +75,68 @@ var ProfileCmd = &cobra.Command{
 		console.Success("minikube profile was successfully set to %s", profile)
 	},
 }
+
+func GetAllProfiles() []string {
+	miniPath := constants.GetMinipath()
+	profilesPath := filepath.Join(miniPath, "profiles")
+	fileInfos, err := ioutil.ReadDir(profilesPath)
+	if err != nil {
+		console.ErrLn("Unable to list in dir: %s \n Error: %v", profilesPath, err)
+	}
+
+	var profiles []string
+	for _, fileInfo := range fileInfos {
+		if fileInfo.IsDir() {
+			profilePath := filepath.Join(profilesPath, fileInfo.Name())
+			if isValidProfile(profilePath) {
+				profiles = append(profiles, fileInfo.Name())
+			}
+		}
+	}
+	return profiles
+}
+
+func isValidProfile(profilePath string) bool {
+	fileInfos, err := ioutil.ReadDir(profilePath)
+	if err != nil {
+		console.ErrLn("Unable to list in dir: %s \n Error: %v", profilePath, err)
+	}
+
+	hasConfigJSON := false
+	for _, fileInfo := range fileInfos {
+		if fileInfo.Name() == "config.json" {
+			hasConfigJSON = true
+		}
+	}
+
+	if !hasConfigJSON {
+		return false
+	}
+
+	// TODO: Use constants?
+	profileConfigPath := filepath.Join(profilePath, "config.json")
+	bytes, err := ioutil.ReadFile(profileConfigPath)
+	if err != nil {
+		console.ErrLn("Unable to read file: %s \n Error: %v", profileConfigPath, err)
+	}
+
+	var configObject mkConfig.Config
+	errUnmarshal := json.Unmarshal(bytes, &configObject)
+
+	if errUnmarshal != nil {
+		console.ErrLn("Could not unmarshal config json to config object: %s \n Error: %v", profileConfigPath, err)
+	}
+	return IsProfileConfigValid(configObject)
+}
+
+func IsProfileConfigValid(configObject mkConfig.Config) bool {
```

Comment:
Maybe leave this part out of this PR. Focus on getting the delete all working, with the basic JSON validation of the file, this validation can be done on following PR, keeping this short to review.